Sources/Testing/EntryPoints/ABIv0/Encoded/ABIv0.EncodedEvent.swift

```diff
@@ -0,0 +1,103 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension ABIv0 {
+  /// A type implementing the JSON encoding of ``Event`` for the ABI entry point
+  /// and event stream output.
+  ///
+  /// This type is not part of the public interface of the testing library. It
+  /// assists in converting values to JSON; clients that consume this JSON are
+  /// expected to write their own decoders.
+  struct EncodedEvent: Sendable {
+    /// An enumeration describing the various kinds of event.
+    ///
+    /// Note that the set of encodable events is a subset of all events
+    /// generated at runtime by the testing library.
+    ///
+    /// For descriptions of individual cases, see ``Event/Kind``.
+    enum Kind: String, Sendable {
+      case runStarted
+      case testStarted
+      case testCaseStarted
+      case issueRecorded
+      case knownIssueRecorded
+      case testCaseEnded
+      case testEnded
+      case testSkipped
+      case runEnded
+    }
+
+    /// The kind of event.
+    var kind: Kind
+
+    /// The source location of the event, if applicable.
+    var sourceLocation: SourceLocation?
+
+    /// The instant at which the event occurred on the current system's
+    /// suspending clock.
+    var timestamp: Double
+
+    /// The instant at which the event occurred on the wall clock.
+    var timestampSince1970: Double
```

Comment:
These two timestamp properties feel like they could be grouped together into some kind of single type, maybe `EncodedInstant` ?

Reply:
Maybe. I originally just had `timestamp` as a single numeric value, but added the wall clock value later. @plemarquand What do you think?

Reply:
I think wrapping them up in an `instant` field would make sense. 

If timestamp is a https://developer.apple.com/documentation/swift/suspendingclock maybe we could link to those docs to make it more clear when clock suspension might happen.